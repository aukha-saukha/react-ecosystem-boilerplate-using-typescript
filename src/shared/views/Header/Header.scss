@import '../../../static/css/_variables';

.header {
  border-bottom: 1px solid map-get($border-color, header-bottom);
  grid-area: header;
}

.logo {
  color: map-get($text-color, logo);
  font-weight: normal;

  &:hover {
    color: map-get($text-color, logo);
  }

  &:visited {
    color: map-get($text-color, logo);
  }
}
