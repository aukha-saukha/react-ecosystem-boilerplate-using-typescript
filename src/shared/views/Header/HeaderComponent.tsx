import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { LOGO } from '../../../data/views-schema/header';
import HeaderStyle from './Header.scss';

const HeaderComponent: React.FC<{}> = (): JSX.Element => (
  <header
    className={`display-flex flex-items-align-center padding-all-15 ${HeaderStyle['header']}`}
  >
    <h1>
      <NavLink
        className={`width-100 ${HeaderStyle['logo']}`}
        data-test="logo"
        to={LOGO.linkToLocation}
      >
        {LOGO.name}
      </NavLink>
    </h1>
  </header>
);

export default HeaderComponent;
