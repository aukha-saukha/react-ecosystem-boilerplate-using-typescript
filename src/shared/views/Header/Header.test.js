import { shallow } from 'enzyme';
import React from 'react';

import Header from './HeaderComponent';

describe('Header component', () => {
  it('renders correctly.', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
