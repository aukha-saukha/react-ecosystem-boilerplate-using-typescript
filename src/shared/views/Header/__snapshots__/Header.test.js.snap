// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly. 1`] = `
<header
  className="display-flex flex-items-align-center padding-all-15 header"
>
  <h1>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="width-100 logo"
      data-test="logo"
      to="/"
    >
      REB
    </NavLink>
  </h1>
</header>
`;
